src/btf.cpp

```diff
@@ -412,6 +412,20 @@ SizedType BTF::get_stype(const BTFId &btf_id, bool resolve_structs)
     stype = CreatePointer(
         get_stype(BTFId{ .btf = btf_id.btf, .id = t->type }, false));
   }
+  else if (btf_is_array(t))
+  {
+    auto *array = btf_array(t);
+    const auto &elem_type = get_stype(
+        BTFId{ .btf = btf_id.btf, .id = array->type });
+    if (elem_type.type == Type::integer && elem_type.GetSize() == 1)
```

Comment:
Help, I'm confused by this if check. If the element type is an integer with size 1 then create string? Is there a test for this case below and I'm just missing it?

Reply:
A `char` is an integer with size 1 (`int8_t`, `uint8_t` etc. are just aliases for char types, so no difference). The "char_arr" member in the new test case covers this.


Doing the conversion in the type parser isn't the nicest way of handling this, but it does work without too many changes. Ideally I think I'd like to parse it as a char-array at this stage and then have an implicit casting system that could convert it into `Type::string` later when needed (e.g. when printing or comparing against a string).

That was too much work for me today though so I copied from our other type parsers:
https://github.com/iovisor/bpftrace/blob/78eebadde8a0000e65c06235ed4c93d5f940907c/src/clang_parser.cpp#L300-L303
https://github.com/iovisor/bpftrace/blob/78eebadde8a0000e65c06235ed4c93d5f940907c/src/dwarf_parser.cpp#L200-L203

Reply:
Ahh thanks for the explanation. I see "type::string" in the tests now for char_arr